### ISSUE TITLE
Reroute discarded events from throttle transform

### ISSUE THREAD

**Issue by fpytloun** [vectordotdev/vector#13549]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Use Cases

I would like to create metric with extra labels from throttled events fields (eg. `tag` or `service`) to be able to see what was throttled.

There might be more use-cases, eg. send throttled events to some other backend for further processing.

### Attempted Solutions

Using `vector_events_discarded_total` metric cannot give any detailed info:
<img width="1360" alt="image" src="https://user-images.githubusercontent.com/948799/178950919-d4e5cac0-97c6-48e8-a492-142e077fb0e3.png">

### Proposal

Might be done by routing throttled events away and attach additional log_to_metric transform on that events. Basically same feature as dropped option for remap transform.

> events that result in runtime errors or aborts will be dropped from the default output stream and sent to the dropped output instead. For a transform component named foo, this output can be accessed by specifying foo.dropped as the input to another component.

**Comment by jszwedko** [comment]
> Responded in Discord as well - but we may be passing just the name of the key, rather than the contained value from that key (which is what _should_ be bucketed).

Yeah, that looks like the case. I'll open a separate bug for that since I think this feature request to be able to route dropped events is also a good one.

**Comment by fpytloun** [comment]
@jszwedko checking related code: https://github.com/vectordotdev/vector/blob/master/src/transforms/throttle.rs#L153

and it indeed seems that `key_field` should be template to work as expected, otherwise key would be static value and basically useless (it would just rate-limit everything) 🤯 
So it's definately wrongly named and wrongly documented (`key` or `throttle_key` would make at least some sense but `key_field` gives impression that it should be just field name).

**Comment by jszwedko** [comment]
👍 I think I like `throttle_key` as an improvement at least.

**Comment by jszwedko** [comment]
@fpytloun just a note that a workaround came up today that would involve using `log_to_metric` both in-front of, and behind, the throttle transform to generate two metrics that could be subtracted to get the metrics about the throttled events.
